frontend/src/components/FacilityBooking/BlockStyles.ts

```diff
@@ -0,0 +1,106 @@
+import styled, { css } from 'styled-components'
+import { TimeBlockType } from '../../store/facilityBooking/types'
+
+export const blockHeight = 90
+export const topDistance = 10
+export const blockGap = 5
+
+export const DailyContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: ${`${topDistance}px`} 5px;
+  gap: ${`${blockGap}px`};
+`
+
+const blockStyles = css`
+  position: relative;
+  border-radius: 10px;
+  padding: 15px;
+  color: #000;
+  display: flex;
+  align-items: center;
+  white-space: nowrap;
+  height: ${`${blockHeight}px`};
+`
+
+// Styles for Booking Block (Occupied, Available, Selected and Not Available)
+export const StyledBookingBlock = styled.div<{
+  type: TimeBlockType
+  blockId: number
+  hasOverlay?: boolean
+}>`
+  ${(props) => (props.type === TimeBlockType.AVAILABLE || props.type === TimeBlockType.SELECTED) && 'cursor: pointer;'}
+  ${blockStyles}
+  background: ${(props) =>
+    props.type === TimeBlockType.OCCUPIED
+      ? palette.occupied
+      : props.type === TimeBlockType.AVAILABLE
+      ? palette.available
+      : props.type === TimeBlockType.SELECTED
+      ? palette.selected
+      : props.blockId % 2 == 0
+      ? palette.unavailable1
+      : palette.unavailable2};
+
+  ${(props) =>
+    props.hasOverlay &&
+    css`
+      &::after {
+        content: '';
+        position: absolute;
+        top: 0;
+        left: 0;
+        right: 0;
+        bottom: 0;
```

Comment:
experimented if this required? like no purpose

Reply:
is a pseudo class, need to use to cover the block itself

Reply:
i have no purpose